python_modules/dagster/dagster/_core/instance/__init__.py

```diff
@@ -2447,7 +2490,11 @@ def handle_new_event(
                 and event.get_dagster_event().is_job_event
             ):
                 self._run_storage.handle_run_event(run_id, event.get_dagster_event())
-
+                run = self.get_run_by_id(run_id)
+                if run and event.get_dagster_event().event_type == DagsterEventType.RUN_FAILURE:
+                    self.add_run_tags(
+                        run_id, {WILL_RETRY_TAG: str(self._should_retry_run(run)).lower()}
```

Comment:
there's a theoretical race condition here that was not there before where two runs from the same run group fail at the exact same time and both have this set to True, where before the daemon would have processed them sequentially and set the first one to True and the second one to False - I don't think 'two runs from the same run group' is a particularly common thing in the first place so this concern may be theoretical